packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -4479,6 +4498,11 @@ export class ContainerRuntime
 			// If we're resubmitting a batch, keep the same "staged" value as before.  Otherwise, use the current "global" state.
 			const staged = this.batchRunner.resubmitInfo?.staged ?? this.inStagingMode;
 
+			assert(
+				!staged || canStageMessageOfType(type),
+				`Unexpected message submitted in staging mode, type=${type}`,
```

Comment:
we can't use format strings in asserts. if you want to get the string in the error you need to throw an error